backtracking/generate_parentheses.cpp

```diff
@@ -0,0 +1,97 @@
+/**
+ * @file
+ * @brief Generates all combinations of well-formed parentheses.
+ *
+ * @details a sequence of parentheses is well-formed if each opening parentheses
+ has a corresponding closing parenthesis
+ *  and the closing parentheses are correctly ordered
+ *
+ * @author [Giuseppe Coco](https://github.com/WoWS17)
+
+ */
+
+#include <cassert>
+#include <iostream>
+#include <vector>
```

Comment:
Please add a one-line description of what the library/header is for (**see the example below**).

```c++
#include <iostream>    /// for IO operations
#include <cassert>    /// for assert
```